src/prefect/server/models/deployments.py

```diff
@@ -977,7 +977,7 @@ async def mark_deployments_not_ready(
         return
 
     async with db.session_context(
-        begin_transaction=True, with_for_update=True
```

Comment:
I think we should remove this. It will lead to more locking of the db, for a comparatively non-critical operation of marking a deployment ready/unready, where we're okay with a little bit of maybe unsteady concurrent updating